db/migrate/20181114213758_create_backup_code_configurations.rb

```diff
@@ -0,0 +1,13 @@
+class CreateBackupCodeConfigurations < ActiveRecord::Migration[5.1]
+  def change
+    create_table :backup_code_configurations do |t|
+      t.integer :user_id, null: false
+      t.string :code_fingerprint, default:'', null: false
+      t.string :encrypted_code, default:'', null: false
+      t.timestamp :used_at
+      t.timestamps
+      t.index [:user_id, :code_fingerprint], name: "index_bcc_on_user_id_code_fingerprint",
```

Comment:
Not a blocker for this PR (I think), but something we'll need to do in the future: make sure we don't generate a code that has the same fingerprint as a code already used, unless we add a `where` clause to the index to scope it to those codes for which `used_at` is `NULL`.